### ISSUE TITLE
Introduce prioirty ordering for different types of processing task

### ISSUE THREAD

**Issue by johnaohara** [Hyperfoil/Horreum#752]
All tasks are currently equal priority. However some tasks will naturally be a higher priority than others i.e. `Blocking Experiment` > `Label Values recalculation` 

I think we should introduce a priority to the data manipulation tasks, so that important tasks are always executed in a timely manner.

**Comment by johnaohara** [comment]
Does processing new datatsets occur in a different thread pool from recalculations?

We need to be able to return some results much faster than others, e.g. a user making a blocking call to determine a pass/fail result needs to be returned in close to real-time as possible, whereas recalculation jobs should occur in the background with low priority and complete in a non-determinate time.

**Comment by johnaohara** [comment]
It looks like new Runs are now calculated on the thread handling the http request, is that correct?

**Comment by stalep** [comment]
Yes, and only recalculating datasets because of labelchanges are pushed to a queue.  Run recalculations are pushed to another separate queue.
All other requests are synchronous.

**Comment by jesperpedersen** [comment]
I think this ok.

We can look at UI and OpenAPI interaction in releases post 0.10.x if serious issues pops up as part of removing the message bus as a central component
